File: Server_Root/Server/RoomServer/MasterData/Behavior.cpp
#include "stdafx.h"
#include "Behavior.h"
#include "../Util.h"

namespace MasterData
{
	Behavior::Behavior()
	{
	}

	Behavior::~Behavior()
	{
	}

	void Behavior::SetData(vector<string> data)
	{
		m_nID = atoi(data[0].c_str());
		m_strName = data[1];
		m_strClassName = data[2];
		m_fLength = atof(data[3].c_str());
		m_strStringParams = data[4];

		vector<string> vecString;
		vector<string> vecString2;
		Util::Parse(data[6], ',', &vecString);
		for (vector<string>::iterator it = vecString.begin(); it != vecString.end(); ++it)
		{
			Util::Parse(*it, ':', &vecString2);

			Action action;
			action.m_strID = vecString2[0];
			action.m_fTime = atof(vecString2[1].c_str());
			action.m_vecParams.resize(vecString2.size());
			copy(vecString2.begin() + 2, vecString2.end(), action.m_vecParams.begin());

			m_vecAction.push_back(action);
		}
	}
}

File: Client_Root/Client/Assets/Scripts/MasterData/Behavior.cs
using System.Collections.Generic;

namespace MasterData
{
    public class Behavior : IMasterData
    {
        public struct Action
        {
            public string m_strID;
            public float m_fTime;
            public List<string> m_listParams;
        };

        public string  m_strName;
        public string  m_strClassName;
        public float   m_fLength;
        public string  m_strStringParams;
        public string  m_strAnimationName;
        public List<MasterData.Behavior.Action> m_listAction = new List<Action>();

        public override void SetData(List<string> data)
        {
            Util.Convert(data[0], ref m_nID);
            m_strName = data[1];
            m_strClassName = data[2];
            Util.Convert(data[3], ref m_fLength);
            m_strStringParams = data[4];
            m_strAnimationName = data[5];

            if (data[6] != "")
            {
                List<string> listString = new List<string>();
                List<string> listString2 = new List<string>();
                Util.Parse(data[6], ',', listString);
                foreach(string str in listString)
                {
                    Util.Parse(str, ':', listString2);

                    Action action;
                    action.m_strID = listString2[0];
                    action.m_fTime = float.Parse(listString2[1]);
                    action.m_listParams = new List<string>();

                    for (int i = 2; i < listString2.Count; ++i)
                    {
                        action.m_listParams.Add(listString2[i]);
                    }

                    m_listAction.Add(action);
                }
            }
        }
    }
}